File: domain/src/main/java/com/everis/domain/interactor/CheckLoggedIn.java
package com.everis.domain.interactor;

import com.everis.domain.executor.PostExecutorThread;
import com.everis.domain.repository.UserRepository;

import rx.Observable;

/**
 * Created by everis on 26/04/18.
 */

public class CheckLoggedIn extends UseCase {

    private UserRepository repository;

    public CheckLoggedIn(PostExecutorThread postExecutorThread, UserRepository repository) {
        super(postExecutorThread);
        this.repository = repository;
    }

    @Override
    protected Observable buildUseCaseObservable() {
        return repository.checkUserLoggedIn();
    }
}

File: domain/src/main/java/com/everis/domain/interactor/CheckLoggedIn.kt
package com.everis.domain.interactor

import com.everis.domain.executor.PostExecutorThread
import com.everis.domain.model.User
import com.everis.domain.repository.UserRepository

import rx.Observable

class CheckLoggedIn(postExecutorThread: PostExecutorThread, private val repository: UserRepository) : UseCase<User>(postExecutorThread) {

    override fun buildUseCaseObservable(): Observable<User> {
        return repository.checkUserLoggedIn()
    }
}
